src/populateExchange.ts

```diff
@@ -153,77 +127,77 @@ interface TypeFragment {
   key: number;
   /** Selection set. */
   fragment: FragmentDefinitionNode;
-  /** Type of selection. */
-  type: string;
-}
-
-interface MakeFragmentsFromQueryArg {
-  schema: GraphQLSchema;
-  query: DocumentNode;
 }
 
 /** Gets typed selection sets and fragments from query */
-export const extractSelectionsFromQuery = ({
-  schema,
-  query,
-}: MakeFragmentsFromQueryArg) => {
-  let selections: { selections: SelectionSetNode; type: string }[] = [];
-  let fragments: FragmentDefinitionNode[] = [];
+export const extractSelectionsFromQuery = (
+  schema: GraphQLSchema,
+  query: DocumentNode
+) => {
+  const extractedFragments: FragmentDefinitionNode[] = [];
+  const newFragments: FragmentDefinitionNode[] = [];
   const typeInfo = new TypeInfo(schema);
 
   visit(
     query,
     visitWithTypeInfo(typeInfo, {
       Field: node => {
-        if (!node.selectionSet) {
-          return undefined;
+        if (node.selectionSet) {
+          const type = getTypeName(typeInfo);
+          newFragments.push({
+            kind: Kind.FRAGMENT_DEFINITION,
+            typeCondition: {
+              kind: Kind.NAMED_TYPE,
+              name: nameNode(type),
+            },
+            name: nameNode(`${type}_PopulateFragment_`),
+            selectionSet: node.selectionSet,
+          });
         }
-
-        const type = getTypeName(typeInfo);
-        selections = [...selections, { selections: node.selectionSet, type }];
       },
       FragmentDefinition: node => {
-        fragments = [...fragments, node];
+        extractedFragments.push(node);
       },
     })
   );
 
-  return { selections, fragments };
+  return [extractedFragments, newFragments];
 };
 
-interface AddFragmentsToQuery {
-  schema: GraphQLSchema;
-  query: DocumentNode;
-  typeFragments: Record<string, Omit<TypeFragment, 'key'>[]>;
-  userFragments: UserFragmentMap;
-}
-
 /** Replaces populate decorator with fragment spreads + fragments. */
-export const addFragmentsToQuery = ({
-  schema,
-  query,
-  typeFragments,
-  userFragments,
-}: AddFragmentsToQuery) => {
+export const addFragmentsToQuery = (
+  schema: GraphQLSchema,
```

Comment:
Is there an advantage to changing these to indexed parameters? It's less transparent as to what the arguments should be.

Reply:
It gets rid of the property names in the output JS (although I'd love if Terser would do that automatically)

I personally think it's still transparent since TS typehints (of the TS language server for instance) still give you the signature with names and types.